apps/namadillo/src/hooks/useProposalIdParam.ts

```diff
@@ -0,0 +1,33 @@
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+import { useSanitizedParams } from "@namada/hooks";
+import GovernanceRoutes from "App/Governance/routes";
+
+const parseProposalId = (asString: string | undefined): bigint | null => {
```

Comment:
Nitpick: I would say we should keep BigNumber to make everything standard / compatibility, but we're not making any calculations with that, so I think it's okay. Any thoughts?

Reply:
I had a further think about this. I think `bigint` is good because it more closely matches the values that are valid proposal IDs (basically `u64` in Rust), whereas `BigNumber` includes more values that are not valid (e.g. non-integers). `bigint` is a bit of pain sometimes to use because it can't be used to index objects or as a React key, but `BigNumber` would have the same problems. If we do need to get a `BigNumber` representation we can always do `BigNumber(someBigInt.toString())`. What do you think? I'll leave it in this PR for now and if you still think it should be changed we can add it later.